filebeat/docs/how-does-filebeat-work.asciidoc

```diff
@@ -0,0 +1,76 @@
+== How does Filebeat work?
+
+This page intends to explain what the key building blocks of Filebeat are and how they work together. The goal is that the available configuration options can be better understood on what they do.
+
+=== What is an prospector?
+
+Filebeat currently supports two `prospector` types: `log` and `stdin`. Each prospector type can be defined multiple times. The responsibility of the prospector is to find all sources it should read from. In case of the log input type it is to find all files on the drive based on the defined paths and start a harvester for each file that was found. Each prospector looks as following:
+
+```
+filebeat.prospectors:
+- input_type: log
+  paths:
+    - /var/log/*.log
+    - /var/path2/*.log
+```
+
+The log prospector is responsible to find all files which were defined under the paths above and check, if for each file a harvester has to be started, if one is already running or the file can be ignored because of some configuration options. Each prospector runs in its own go routine and is responsible to manage the harvesters. New files are only picked up, if the offset / size of the file changed since the harvester was closed.
+
+=== What is a harvester?
+
+A harvester is responsible to read the content of a single file. That is done by reading each file line by line and send the information to the output. Each harvester is started by its prospector and retrieves all configuration options from the prospector. The harvester is responsible to open and close files. That means, as long as a harvester is running, the file descriptor stays open. Even if a file is removed or renamed, filebeat will keep reading the file. This has the side affect that the space on your disk will be reserved until the harvester is stopped. Stopping a harvester means:
+
+* The file handler is closed
```

Comment:
Maybe "The file handler is closed, freeing up the underlying resources if the file was deleted".